23-input-scanning/exercises/02-unique-words/main.go

```diff
@@ -32,4 +38,24 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	var (
+		in          = bufio.NewScanner(os.Stdin)
+		totalUnique int
+		total       int
+		unique      = map[string]bool{}
+	)
+
+	in.Split(bufio.ScanWords)
+	for in.Scan() {
+		word := in.Text()
+		_, ok := unique[word]
+		if ok {
+			totalUnique--
+		}
+
+		totalUnique++
+		total++
+		unique[word] = true
```

Comment:
try to keep map (or any structure) operations as close as possible so that the code is coherent

Reply:
fixed